lms/resources/lti_launch.py

```diff
@@ -2,15 +2,23 @@
 import functools
 import hashlib
 
-from pyramid.httpexceptions import HTTPBadRequest
 from pyramid.security import Allow
 
 from lms.resources._js_config import JSConfig
 from lms.values import HUser
 
 
 class LTILaunchResource:
-    """Context resource for LTI launch requests."""
+    """
+    Context resource for LTI launch requests.
+
+    Many methods and properties of this class are only meant to be called when
+    request.parsed_params holds validated params from an LTI launch request and
+    might crash otherwise. So you should only call these methods after the
+    request has been validated with BasicLTILaunchSchema or similar (for
+    example from views that have schema=BasicLTILaunchSchema in their view
+    config).
+    """
```

Comment:
I was wondering if there's anyway to tell this, but presumably:

* It would fail super spectacularly anyway as `parsed_params` will be missing
* We can't really do this in the `__init__()` as the schema won't have been applied yet
* Anything that was at a call level method would just basically re-introduce the `get_params` thing we were trying to get rid of